logger_block.py

```diff
@@ -33,11 +35,16 @@ def process_signals(self, signals):
             None
         """
         log_func = self._get_logger()
-        for s in signals:
-            try:
-                log_func(s)
-            except:
-                self.logger.error("Failed to log signal")
+
+        try:
+            if self.log_as_list():
+                log_func([signal.to_dict() for signal in signals])
+            else:
+                for s in signals:
+                    log_func(s.to_dict())
+        except:
+            self.logger.exception("Failed to log {} signals"
```

Comment:
This isn't entirely true. If we're logging one signal at a time and it's not the first one that fails, then this number is incorrect.